File: perl/22_fibonacci_series.pl
#!/usr/bin/env perl

use strict;
use warnings;

sub fibonacci {
	my $n = shift @_;

	return 0 if $n == 0;
	return 1 if $n == 1;

	return fibonacci( $n - 1 ) + fibonacci( $n - 2 );
}

open my $fh, '<', $ARGV[0];

for my $line ( <$fh> ) {
	print fibonacci( $line ) . "\n";
}

close $fh;

File: python/22_fibonacci_series.py
import sys

def fibonacci(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1

    return fibonacci(n - 1) + fibonacci(n - 2)

with open(sys.argv[1]) as fh:
    for line in fh.readlines():
        line = line.strip()
        if line == '':
            continue

        print fibonacci(int(line))

